mutex_protected.h

```diff
@@ -34,6 +35,14 @@ class mutex_protected {
 
   mutex_locked<T> lock() { return mutex_locked(mutex, &v); }
 
+  std::optional<mutex_locked<T>> try_lock() {
+    if (mutex.try_lock()) {
+      return mutex_locked(mutex, &v);
```

Comment:
I fear another process could acquire the lock in between checking and locking

Reply:
Oh, ok, this is just broken then, since try_lock actually acquires the lock, and then tries to acquire it again in the lock guard.